File: challenge_C/level2/멀리 뛰기.c
#include <stdio.h>

int arr[2001] = {0,};

int DP(int n)
{
    arr[1] = 1;
    arr[2] = 2;
    
    for(int i = 3; i <= n; i++)
        arr[i] = (arr[i - 2] + arr[i - 1]) % 1234567;
    return arr[n];
}

long long solution(int n)
{
    return DP(n);
}

File: challenge_Java/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        if (n == 1) return 1;
        
        int[] jump = new int[n];
        jump[0] = 1;
        jump[1] = 2;
        for (int i = 2; i < n; i++)
            jump[i] = (jump[i - 1] + jump[i - 2]) % 1234567;
        
        return jump[n - 1];
    }
}
